### PR TITLE
added input model optimization

### PR DESCRIPTION
**Pull request by Justinyu1618**

### CODE REVIEW

File: src/algos/simba_algo.py [hunk 1]
```diff
@@ -30,6 +32,12 @@ def train(self):
     def eval(self):
         self.mode = "val"
         self.client_model.eval()
+
```
Comment (Justinyu1618) [inline src/algos/simba_algo.py:38]: added these two methods to enable/disable logs, and enable/disable whether the output is detached (in the case of model_optimization where we need the gradients)

File: src/configs/system_config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "dataset_path": "/u/abhi24/Datasets/Faces/fairface/",
-    "experiments_folder": "/u/abhi24/Workspace/simba/experiments/",
-    "gpu_devices": [2,3]
+  "dataset_path": "/home/justinyu/fairface/",
+  "experiments_folder": "/home/justinyu/experiments/",
+  "gpu_devices": [1, 3]
```
Comment (Justinyu1618) [inline src/configs/system_config.json:4]: reminder: revert this before merge

**Review by tremblerz — APPROVED** [review]
Looks all good! Thanks
